cmd/query/app/querysvc/adjuster/sort.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package adjuster
+
+import (
+	"sort"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+var _ Adjuster = (*SortAttributesAndEventsAdjuster)(nil)
+
+// SortAttributesAndEvents creates an adjuster that standardizes trace data by sorting elements:
+// - Resource attributes are sorted lexicographically by their keys.
+// - Span attributes are sorted lexicographically by their keys.
+// - Span events are sorted lexicographically by their names.
+// - Attributes within each span event are sorted lexicographically by their keys.
+func SortAttributesAndEvents() SortAttributesAndEventsAdjuster {
+	return SortAttributesAndEventsAdjuster{}
+}
+
+type SortAttributesAndEventsAdjuster struct{}
+
+func (s SortAttributesAndEventsAdjuster) Adjust(traces ptrace.Traces) {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		rs := resourceSpans.At(i)
+		resource := rs.Resource()
+		s.sortAttributes(resource.Attributes())
+		scopeSpans := rs.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			ss := scopeSpans.At(j)
+			spans := ss.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				s.sortAttributes(span.Attributes())
+				s.sortEvents(span.Events())
```

Comment:
@yurishkuro Wanted a sanity check here. In v1, the "event" field was moved to the first position (ref: https://github.com/jaegertracing/jaeger/blob/main/model/adjuster/sort_tags_and_log_fields.go#L31). However, in v2, logs are stored as event attributes and the "event" attribute is stored as the event name (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/jaeger/jaegerproto_to_traces.go#L406).

Reply:
sgtm. Good to have first class fields.